tests/baselines/reference/importDeclFromTypeNodeInJsSource.js

```diff
@@ -0,0 +1,40 @@
+//// [tests/cases/compiler/importDeclFromTypeNodeInJsSource.ts] ////
+
+//// [index.d.ts]
+/// <reference path="events.d.ts" />
+//// [events.d.ts]
+declare module "events" {
+    namespace EventEmitter {
+        class EventEmitter {
+            constructor();
+        }
+    }
+    export = EventEmitter;
+}
+
+//// [b.js]
+import { EventEmitter } from 'events';
+class Foo extends EventEmitter {
+    constructor() {
+        super();
+    }
+}
+export default Foo;
+
+//// [b.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const events_1 = require("events");
+class Foo extends events_1.EventEmitter {
+    constructor() {
+        super();
+    }
+}
+exports.default = Foo;
+
+
+//// [b.d.ts]
+export default Foo;
+declare class Foo extends EventEmitter {
+}
+import { EventEmitter } from "events";

```

Comment:
I see, so the key point to this test is that this `.d.ts` doesn't turn into `node/events` because it's a special case global 👍🏻

Reply:
This fix seems incorrect and an odd special case. We expect in cases like this one, to use the module name specified by the ambient module declaration, _not_ one from a filepath. Since the module is declared as `declare module "events"` we should only be using `"events"` to refer to it, and should never have looked at a filepath to begin with.

Reply:
This change needs to be reverted and correctly fixed as @weswigham points out correctly.

Reply:
Oh.... Sorry for the misleading...
And, I am a little confused, if there is not any special folder, according to the module resolution https://www.typescriptlang.org/docs/handbook/module-resolution.html. How could file find the correct modules? Did I miss something?

Reply:
Auh, I find this https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html#-reference-ing-a-module
So, I guess the expected code is not just
```
export default Foo;
declare class Foo extends EventEmitter {
}
import { EventEmitter } from "events";
```

but
```
/// <reference types="node" />
export default Foo;
declare class Foo extends EventEmitter {
}
import { EventEmitter } from "events";
```
Right?

Reply:
https://github.com/microsoft/TypeScript/pull/41895 this one!